File: Excel/474 Challenge.py
import pandas as pd

input = pd.read_excel("474 Wavy Numbers.xlsx", usecols="A", nrows=10)
test = pd.read_excel("474 Wavy Numbers.xlsx", usecols="B", nrows=5)

def is_wavy(number):
    digits = [int(d) for d in str(number)]
    differences = [digits[i+1] - digits[i] for i in range(len(digits)-1)]
    signs = [1 if diff > 0 else -1 if diff < 0 else 0 for diff in differences]
    if len(signs) < 2:
        return False
    return all(abs(signs[i+1] - signs[i]) == 2 for i in range(len(signs)-1))

result = input[input['Numbers'].apply(is_wavy)][['Numbers']]\
    .rename(columns={'Numbers': 'Answer Expected'})\
    .reset_index(drop=True)

print(result.equals(test)) # True

File: Excel/400-499/474/474 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/474 Wavy Numbers.xlsx", range = "A1:A10")
test  = read_excel("Excel/474 Wavy Numbers.xlsx", range = "B1:B6")

is_wavy <- function(number) {
  digits <- str_split(as.character(number), "")[[1]] %>% as.numeric()
  differences <- diff(digits)
  signs <- sign(differences)
  if (length(signs) < 2) {
    return(FALSE)
  }
  all(abs(diff(signs)) == 2)
}

result = input %>%
  mutate(wavy = map_lgl(Numbers, is_wavy)) %>%
  filter(wavy) %>%
  select(`Answer Expected` = Numbers)

identical(result, test)
# [1] TRUE